app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CreateEventActivity.kt

```diff
@@ -86,16 +95,46 @@ class CreateEventActivity : ActivityForAccessingTumOnline<CreateEventResponse>(R
         setDateAndTimeListeners()
 
         createEventButton.setOnClickListener {
-            if (end.isBefore(start)) {
+            if (end.isBefore(start) || (repeats && last.isBefore(start))) {
                 showErrorDialog(getString(R.string.create_event_time_error))
                 return@setOnClickListener
             }
+            if (repeats) {
+                if (endAfterRadioBtn.isChecked && (eventRepeatsTimes.text.toString() == "" ||
+                        eventRepeatsTimes.text.toString().toInt() < 2)) {
+                    showErrorDialog(getString(R.string.create_event_too_little_error))
+                    return@setOnClickListener
+                }
+                // Don't allow too many requests
+                if ((endOnRadioBtn.isChecked && start.plusMonths(6).isBefore(last)) ||
+                        (endAfterRadioBtn.isChecked && eventRepeatsTimes.text.toString().toInt() > 25)) {
+                    showErrorDialog(getString(R.string.create_event_too_many_error))
+                    return@setOnClickListener
+                }
+            }
             if (isEditing) {
                 editEvent()
             } else {
                 createEvent()
             }
         }
+
+        // edited events cannot repeat
+        if (isEditing) {
+            repeatingSwitch.layoutParams = LinearLayout.LayoutParams(0, 0)
```

Comment:
Is `LinearLayout.LayoutParams(0, 0)` a standard practice? If not, please add a comment about your intent, so its understandable what you wanted to happen.